File: Finanzuebersicht.Backend.Admin.Core/Contract/Logic/Modules/AdminUserManagement/AdminUserGroups/DTOs/IAdminUserGroupDetail.cs
using Finanzuebersicht.Backend.Admin.Core.Contract.Logic.Modules.AdminUserManagement.AdminAdGroups;
using Finanzuebersicht.Backend.Admin.Core.Contract.Logic.Modules.AdminUserManagement.AdminAdUsers;
using Finanzuebersicht.Backend.Admin.Core.Contract.Logic.Modules.AdminUserManagement.AdminEmailUsers;
using Finanzuebersicht.Backend.Admin.Core.Contract.Logic.Modules.AdminUserManagement.Permissions;
using System;
using System.Collections.Generic;

namespace Finanzuebersicht.Backend.Admin.Core.Contract.Logic.Modules.AdminUserManagement.AdminUserGroups
{
    public interface IAdminUserGroupDetail
    {
        Guid Id { get; set; }

        string Name { get; set; }

        IDictionary<string, PermissionStatus> Permissions { get; set; }

        IDictionary<string, PermissionStatus> CalculatedPermissions { get; set; }

        IEnumerable<IAdminUserGroup> ParentAdminUserGroups { get; set; }

        IEnumerable<IAdminEmailUser> AdminEmailUserMembers { get; set; }

        IEnumerable<IAdminUserGroup> AdminUserGroupMembers { get; set; }

        IEnumerable<IAdminAdUser> AdminAdUserMembers { get; set; }

        IEnumerable<IAdminAdGroup> AdminAdGroupMembers { get; set; }
    }
}

File: Finanzuebersicht.Frontend.Web.Admin/src/app/model/admin-user-management/admin-user-groups/dtos/i-admin-user-group-detail.ts
import { fromApiAdminAdGroup, IAdminAdGroup } from '../../admin-ad-groups/dtos/i-admin-ad-group';
import { fromApiAdminAdUser, IAdminAdUser } from '../../admin-ad-users/dtos/i-admin-ad-user';
import { fromApiAdminEmailUser, IAdminEmailUser } from '../../admin-email-users/dtos/i-admin-email-user';
import { fromApiPermissions, IPermissions } from '../../permissions/dtos/i-permissions';
import { ApiAdminUserGroupDetail } from './api/api-admin-user-group-detail';
import { fromApiAdminUserGroup, IAdminUserGroup } from './i-admin-user-group';

export interface IAdminUserGroupDetail {
    id: string;
    name: string;
    permissions: IPermissions;
    calculatedPermissions: IPermissions;
    parentAdminUserGroups: IAdminUserGroup[];
    adminEmailUserMembers: IAdminEmailUser[];
    adminUserGroupMembers: IAdminUserGroup[];
    adminAdUserMembers: IAdminAdUser[];
    adminAdGroupMembers: IAdminAdGroup[];
}

export function fromApiAdminUserGroupDetail(apiAdminUserGroupDetail: ApiAdminUserGroupDetail): IAdminUserGroupDetail {
    return {
        id: apiAdminUserGroupDetail.id,
        name: apiAdminUserGroupDetail.name,
        permissions: fromApiPermissions(apiAdminUserGroupDetail.permissions),
        calculatedPermissions: fromApiPermissions(apiAdminUserGroupDetail.calculatedPermissions),
        parentAdminUserGroups: apiAdminUserGroupDetail.parentAdminUserGroups
            .map(apiAdminUserGroup => fromApiAdminUserGroup(apiAdminUserGroup)),
        adminEmailUserMembers: apiAdminUserGroupDetail.adminEmailUserMembers
            .map(apiAdminEmailUserMember => fromApiAdminEmailUser(apiAdminEmailUserMember)),
        adminUserGroupMembers: apiAdminUserGroupDetail.adminUserGroupMembers
            .map(apiAdminUserGroup => fromApiAdminUserGroup(apiAdminUserGroup)),
        adminAdUserMembers: apiAdminUserGroupDetail.adminAdUserMembers
            .map(apiAdminAdUserMember => fromApiAdminAdUser(apiAdminAdUserMember)),
        adminAdGroupMembers: apiAdminUserGroupDetail.adminAdGroupMembers
            .map(apiAdminAdGroupMember => fromApiAdminAdGroup(apiAdminAdGroupMember)),
    };
}
